fuzz/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+use fedimint_core::encoding::{Decodable, Encodable};
+
+pub fn test_decodable<T>(data: &[u8])
+where
+    T: Decodable + Encodable,
+{
+    if let Ok(v) = T::consensus_decode(&mut &data[..], &Default::default()) {
+        let encoded_vec = v.consensus_encode_to_vec();
+        // helps debugging to have it standalone
+        let encoded = encoded_vec.as_slice();
+
+        assert_eq!(&data[..encoded.len()], encoded);
```

Comment:
@elsirion As things are `consensus_decode` will not fail on extra leftover input... . Which is fine when they are used internally, but on e2e case it's probably not what the user wants and could lead to some serious bugs.

Will add to #4395

Reply:
I did add checks against that in some places I remember, but surely forgot it in others.

Reply:
@elsirion I looked into it, and it's not possible to do it on `Encodable` trait level, because `Read` does not have a fixed length known in advance. So it always need to be done on the level that holds `&[u8]` or something like that.

Reply:
Yes, does it make sense otherwise? I was thinking of the case where we have a `Cursor<&[u8]>` and can make sure that the cursor was read till the end (at the callsite of decode).